tests/integration/test_category_destpath.py

```diff
@@ -0,0 +1,83 @@
+"""Test if section indexes avoid pages."""
+
+import os
+
+import pytest
+
+import nikola.plugins.command.init
+from nikola import __main__
+from nikola.utils import makedirs
+
+from .helper import append_config, cd, create_simple_post
+from .test_empty_build import (  # NOQA
+    test_archive_exists,
+    test_avoid_double_slash_in_rss,
+    test_check_files,
+    test_check_links,
+    test_index_in_sitemap,
+)
+
+
+def test_destpath_with_avoidance(build, output_dir):
+    """Test destpath categories page generation and avoidance."""
+
+    def _make_output_path(dir, name):
+        """Make a file path to the output."""
+        return os.path.join(dir, name + ".html")
+
+    cat1 = os.path.join(output_dir, "posts", "cat1")
+    cat2 = os.path.join(output_dir, "posts", "cat2")
+
+    # Do all files exist?
+    assert os.path.isfile(_make_output_path(cat1, "index"))
+    assert os.path.isfile(_make_output_path(cat2, "index"))
+
+    # Are their contents correct?
+    with open(_make_output_path(cat1, "index"), "r", encoding="utf-8") as fh:
```

Comment:
Instead of re-creating the same index again I'd suggest storing the path to the cat1 index in a variable and refer to it here and in the file existance test above.